### PR TITLE
go: zookeeper load generator is added

### PR DESCRIPTION
**Pull request by mehmetalisavas**

### CODE REVIEW

**Comment by cihangir** [comment]
is this completed?

**Comment by mehmetalisavas** [comment]
@cihangir yes, i forgot to mention you about PR

File: scripts/generator.go [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+	for i := 0; i < *goroutines; i++ {
+		go consume(c, &wg)
+	}
+
+	<-done
+	wg.Wait()
+
+	if producer == consumer {
+		fmt.Println(fmt.Sprintf("producer:%d and consumer:%d checking is finihed as successfully", producer, consumer))
+	}
```
Comment (cihangir) [inline scripts/generator.go:74]: what about the else case :)

File: scripts/generator.go [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+	address    = flag.String("address", "0.0.0.0", "addresses for zookeeper")
+)
+
+var mu = &sync.Mutex{}
+
+func main() {
+	flag.Parse()
+
+	if *goroutines > *element {
+		fmt.Printf("%v", fmt.Errorf("err: goroutines should be less than element number"))
```
Comment (cihangir) [inline scripts/generator.go]: you could use log.Fatal(...) here too

File: scripts/generator.go [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+	defer wg.Done()
+
+	for {
+		msg, ok := <-msgs
+		if !ok {
+			return
+		}
+		_, err := c.Create(msg, nil, 0, zk.WorldACL(zk.PermAll))
+		if err != nil {
+			if err != zk.ErrNodeExists {
```
Comment (cihangir) [inline scripts/generator.go]: did you mean `err == zk.ErrNodeExists` ?

File: scripts/generator.go [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+	if err != nil {
+		return err
+	}
+
+	for _, child := range children {
+		// delete operation takes path parameter
+		// then it should have '/' as first index
+		if err := c.Delete("/"+child, 0); err != nil {
+			// we can't delete /zookeeper path from the znodes
+			if child == "zookeeper" {
```
Comment (cihangir) [inline scripts/generator.go]: if child is zookeeper, do not try to delete, since it is not deletable

File: scripts/generator.go (hunk 3 above)
Comment (mehmetalisavas) [inline scripts/generator.go]: yeah :) im gonna ignore this err here

**Pull request merged**